### PR TITLE
dev: accelerate tests

### PR DESCRIPTION
**Pull request by greged93**
# Pull Request type

Time spent on this PR: 2 days

Resolves: #568 

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [x] Testing

## What is the new behavior?
Instead of using a macro and storing all the generated tests in one file, the tests are separated into the same folder structure as the ef-tests data using a build script. Additionally, `par_iter` from `rayon` crate is moved from folder iteration to test case iteration, which lowers the change for bottlenecks creation and speeds up compilation time.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: crates/build-utils/src/filter.rs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    pub fn diff(&self, rhs: &Self) -> Vec<Folder> {
+        let mut diff = Vec::new();
+        diff.append(&mut map_diff(&self.filename, &rhs.filename));
+        diff.append(&mut map_diff(&self.regex, &rhs.regex));
+        diff.append(&mut map_diff(&self.test_name, &rhs.test_name));
+        diff
+    }
+}
+
+fn map_diff(lhs: &FilterMap, rhs: &FilterMap) -> Vec<Folder> {
```
Comment (ClementWalter) [inline crates/build-utils/src/filter.rs:72]: what is lhs and rhs?

File: crates/ef-testing/build.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+lazy_static! {
+    static ref SUITE_PATH: PathBuf = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("./ethereum-tests/BlockchainTests/GeneralStateTests");
+    static ref INTEGRATION_TESTS_PATH: PathBuf =
+        PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("./tests/");
+}
+
+fn main() {
+    // Tell Cargo that if the given file changes -> to rerun this build script.
+    println!("cargo:rerun-if-changed={SKIPPED_TESTS}");
```
Comment (ClementWalter) [inline crates/ef-testing/build.rs:22]: maybe "cleaner" to use the macro directly

`rerun_if_changed!(SKIPPED_TESTS)`

https://docs.rs/cargo-emit/latest/cargo_emit/macro.rerun_if_changed.html

**Review by ClementWalter — DISMISSED** [review]
minor suggestions

File: crates/build-utils/src/filter.rs (hunk 1 above)
Comment (greged93) [reply crates/build-utils/src/filter.rs:72]: left hand side and right hand side

File: crates/ef-testing/build.rs (hunk 2 above)
Comment (greged93) [reply crates/ef-testing/build.rs:22]: ah interesting, did not know you could use the macro

File: crates/build-utils/src/filter.rs (hunk 1 above)
Comment (greged93) [reply crates/build-utils/src/filter.rs:72]: hahaha I have no clue what latin hypercube sampling is, but lhs and rhs is quite common in rust when doing [operations](https://doc.rust-lang.org/std/ops/trait.Mul.html).

**Review by ClementWalter — APPROVED** [review]

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**